### PR TITLE
Updated admin's XRegExp to 5.1.1.

### PR DESCRIPTION
**Pull request by AndreasDickow**
Update of XRegExp in Django Admins Urlify function to the recent version

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Comment by felixxm** [comment]
@AndreasDickow Thanks :+1: `XRegExp` 4 and 5 introduce a lot of breaking changes, Are we not affected? :thinking: 

Please also revert unrelated changes to `.gitignore` and `package.json`.

**Comment by felixxm** [comment]
Do we need `xregexp-all.js` (maybe `xregexp.js` will suffice)? We use it in one place:

https://github.com/django/django/blob/594fcc2b7427f7baf2cf1a2d7cd2be61467df0c3/django/contrib/admin/static/admin/js/urlify.js#L159

**Comment by AndreasDickow** [comment]
> Do we need `xregexp-all.js` (maybe `xregexp.js` will suffice)? We use it in one place:
> 
> https://github.com/django/django/blob/594fcc2b7427f7baf2cf1a2d7cd2be61467df0c3/django/contrib/admin/static/admin/js/urlify.js#L159

local tests confirm, that all urlify related Javascript Unit Tests still work if I leave away these addons in the build process of the module:

var _build = _interopRequireDefault(require("./addons/build"));
var _matchrecursive = _interopRequireDefault(require("./addons/matchrecursive"));
var _unicodeProperties = _interopRequireDefault(require("./addons/unicode-properties"));
var _unicodeScripts = _interopRequireDefault(require("./addons/unicode-scripts"));

these two addons had to remain, else tests would have failed:
var _unicodeBase = _interopRequireDefault(require("./addons/unicode-base"));
var _unicodeCategories = _interopRequireDefault(require("./addons/unicode-categories"));

This way I could save 70kb of uncompressed js size, resulting in 160kb .min.js file at the end

File: docs/ref/databases.txt [hunk 1]
```diff
@@ -1179,7 +1179,6 @@ by 3rd parties that allow you to use other databases with Django:
 * :pypi:`Firebird <django-firebird>`
 * :pypi:`Google Cloud Spanner <django-google-spanner>`
 * :pypi:`Microsoft SQL Server <mssql-django>`
-* :pypi:`Snowflake <django-snowflake>`
```
Comment (felixxm) [inline docs/ref/databases.txt:1182]: Please revert this unrelated change.

**Review by felixxm — COMMENTED** [review]
@AndreasDickow Thanks :+1: Please add a release note (`5.0.txt`), e.g.
```
* ``XRegExp`` is upgraded from version 3.2.0 to 5.1.1.
```

File: docs/releases/5.0.txt [hunk 2]
```diff
@@ -504,3 +504,5 @@ to remove usage of these features.
 * Passing ``nulls_first=False`` or ``nulls_last=False`` to ``Expression.asc()``
   and ``Expression.desc()`` methods, and the ``OrderBy`` expression is no
   longer allowed.
+
+* ``XRegExp`` is upgraded from version 3.2.0 to 5.1.1.
```
Comment (felixxm) [inline docs/releases/5.0.txt:508]: This should be in the `` :mod:`django.contrib.admin` `` section.

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Comment by felixxm** [comment]
@AndreasDickow Thanks :+1: Welcome aboard :boat:

**Comment by felixxm** [comment]
buildbot, test on selenium.
